### PR TITLE
Runtime: fixes RTG power, adjusts pipenet, adds debug chem dispensers

### PR DESCRIPTION
**Pull request by 81Denton**
:cl: Denton
fix: Runtimestation's RTGs are now connected to the powernet.
code: Added chem_dispenser/fullupgrade as well as chem_synthesizer for testing.
tweak: Replaced runtimestation's two chem dispensers with those subtypes.
/:cl:

- Runtimestation's RTGs weren't even connected to the power grid, whoops
- Some vents were underneath tables, I moved them
- The part that people might care about: I added a /fullupgrade subtype for chem dispensers that has all stock parts upgraded. Also, chem_synthesizer which has the functionality of SCP-294. It's much faster and more convenient to use this one instead of VV.
- Those two subtypes are only on runtimestation.

### CODE REVIEW

**Comment by arsserpentarium** [comment]
need to add handheld debug chem dispencer.It's more usable, than VV.

**Comment by AverageJoe82** [comment]
I knew those rtgs weren’t connected

**Comment by 81Denton** [comment]
@Mapdiffbot

**Comment by 81Denton** [comment]
@arsserpentarium Something like a portable chem dispenser where you can just input the reagent ID would be great, yeah.

@Cyberboss Looks like @mapdiffbot broke.

**Comment by AlexTheSysop42** [comment]
Put an infinite cell in the debug one

**Comment by 81Denton** [comment]
Good idea

**Comment by 81Denton** [comment]
Alright, I axed chem_dispenser/full/debug and instead added chem_synthesizer. It has the functionality of SCP-294, but again, is only for testing and shouldn't ever leave runtimestation.
Typing chem IDs into a GUI is much faster and more convenient than varediting.

File: code/modules/reagents/chemistry/machinery/chem_synthesizer.dm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+				input_reagent = find_reagent(input_reagent)
+			if(!input_reagent || !GLOB.chemical_reagents_list[input_reagent])
+				say("OUT OF RANGE")
+				return
+			else
+				if(!beaker)
+					return
+				else if(!beaker.reagents && !QDELETED(beaker))
+					beaker.create_reagents(beaker.volume)
+				beaker.reagents.add_reagent(input_reagent, amount)
```
Comment (SpaceManiac) [inline code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: Always adding only ten units seems a little inflexible, maybe make it adjustable?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/machinery/chem_synthesizer.dm (hunk 1 above)
Comment (81Denton) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: Good idea, let me try to mess around with TGUI.

Comment (81Denton) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: Alright this will take a while

Comment (81Denton) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: I've managed to get TGUI to work, but entering a value doesn't actually change the chem synthesizer's `amount` var. Does anyone have an idea?

Comment (SpaceManiac) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: Take a look at what the ORM does, roughly like:

```html
<input value={{amount}} placeholder='###' class="number"/>
<ui-button class='center' action='dispense' state={{(amount >= 1) ? null : 'disabled'}} params='{ "amount" : {{amount}} }' >
```

Maybe use it for the chem ID too if you can pull it off? Just a suggestion.

Comment (81Denton) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: Thanks for the suggestion, I'll give it a shot.

Comment (81Denton) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: @SpaceManiac I realized that I need a whole bunch of practice until I can make it work. If you're fine with merging the PR as it is right now, I'll figure out a way to implement changing the amount eventually.

Comment (SpaceManiac) [reply code/modules/reagents/chemistry/machinery/chem_synthesizer.dm]: It's VVable if nothing else.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**